src/orchestrator.ts

```diff
@@ -177,6 +177,10 @@ export class Orchestrator {
                 decisionStartedEvent = newDecisionStartedEvent || decisionStartedEvent;
                 context.df.currentUtcDateTime = this.currentUtcDateTime = new Date(decisionStartedEvent.Timestamp);
 
+                if (state[partialResult.completionIndex] !== undefined) {
```

Comment:
This is a great simple fix. I'm 100% positive it behaves the exact same as it does in the C#, but I believe it does in most cases, and is a vast improvement over the current behavior.

I would recommend adding a comment for this chunk of code, something like:
```
// The first time a task is marked as complete, the history event that finally marked the task as completed 
// should not yet have been played by the Durable Task framework, resulting in isReplaying being false. 
// On replays, the event will have already been processed by the framework, and IsPlayed will be marked as true.

Reply:
Thanks for your suggestion. I've added the comment.